File: js/bullet.js
import { BULLET_SIZE, BULLET_SPEED, CANVAS_WIDTH, CANVAS_HEIGHT, DIR_UP, DIR_DOWN, DIR_LEFT, DIR_RIGHT, COLORS } from './constants.js';

export class Bullet {
    constructor(x, y, direction, owner) {
        this.x = x - BULLET_SIZE / 2;
        this.y = y - BULLET_SIZE / 2;
        this.width = BULLET_SIZE;
        this.height = BULLET_SIZE;
        this.direction = direction;
        this.speed = BULLET_SPEED;
        this.owner = owner;
        this.isActive = true;
    }

    update() {
        switch (this.direction) {
            case DIR_UP:
                this.y -= this.speed;
                break;
            case DIR_DOWN:
                this.y += this.speed;
                break;
            case DIR_LEFT:
                this.x -= this.speed;
                break;
            case DIR_RIGHT:
                this.x += this.speed;
                break;
        }
    }

    isOutOfBounds() {
        return this.x < 0 || this.x > CANVAS_WIDTH ||
               this.y < 0 || this.y > CANVAS_HEIGHT;
    }

    getBounds() {
        return {
            x: this.x,
            y: this.y,
            width: this.width,
            height: this.height
        };
    }

    destroy() {
        this.isActive = false;
    }

    draw(ctx) {
        ctx.fillStyle = COLORS.BULLET;
        ctx.fillRect(this.x, this.y, this.width, this.height);
    }
}

File: src/bullet.py
import pygame
from src.constants import UP, DOWN, LEFT, RIGHT
import config


class Bullet:
    def __init__(self, x, y, direction, owner):
        self.x = x
        self.y = y
        self.direction = direction
        self.speed = config.BULLET_SPEED
        self.owner = owner
        self.damage = 1
        self.is_active = True
        self.size = config.BULLET_SIZE
    
    def update(self):
        if self.direction == UP:
            self.y -= self.speed
        elif self.direction == DOWN:
            self.y += self.speed
        elif self.direction == LEFT:
            self.x -= self.speed
        elif self.direction == RIGHT:
            self.x += self.speed
        
        self.check_boundary()
    
    def check_boundary(self):
        if (self.x < 0 or self.x > config.SCREEN_WIDTH or 
            self.y < 0 or self.y > config.SCREEN_HEIGHT):
            self.is_active = False
    
    def draw(self, screen):
        pygame.draw.circle(screen, config.BULLET_COLOR, 
                         (int(self.x), int(self.y)), self.size)
    
    def get_rect(self):
        return pygame.Rect(self.x - self.size, self.y - self.size, 
                          self.size * 2, self.size * 2)
